simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
+                k_point / (np.linalg.norm(k_vector))
+                for k_vector, k_point in zip(reciprocal_lattice_vectors, self.grid)
+            ]
+        )
+
+    def resolve_k_line_density(
+        self, model_systems: List[ModelSystem], logger: BoundLogger
+    ) -> Optional[pint.Quantity]:
+        """
+        Resolves the `k_line_density` of the `KMesh` from the the list of `ModelSystem`.
+
+        Args:
+            model_systems (List[ModelSystem]): The list of `ModelSystem` sections.
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[pint.Quantity]): The resolved `k_line_density` of the `KMesh`.
+        """
+        for model_system in model_systems:
+            # General checks to proceed with normalization
+            if not model_system.is_representative:
+                logger.warning(
+                    "The last `ModelSystem` was not found to be representative."
+                )
+                return None
+            if model_system.type != "bulk":
+                logger.warning("`ModelSystem.type` is not describing a bulk system.")
+                return None
+            atomic_cell = model_system.atomic_cell
+            if atomic_cell is None:
+                logger.warning("`ModelSystem.atomic_cell` was not found.")
+                return None
+
+            # Set the `reciprocal_lattice_vectors` using ASE
+            ase_atoms = atomic_cell[0].to_ase_atoms(logger)
+            if self.reciprocal_lattice_vectors is None:
+                self.reciprocal_lattice_vectors = (
+                    2 * np.pi * ase_atoms.get_reciprocal_cell() / ureg.angstrom
+                )
+
+            # Resolve `k_line_density`
+            if k_line_density := self.get_k_line_density(
+                self.reciprocal_lattice_vectors, logger
+            ):
+                return k_line_density * ureg("m")
+        return None
+
+    def normalize(self, archive, logger) -> None:
+        super().normalize(archive, logger)
+
+        # Set the name of the section
+        self.name = self.m_def.name if self.name is None else self.name
+
+        # If `grid` is not defined, we do not normalize the KMesh
+        if self.grid is None:
+            logger.warning("Could not find `KMesh.grid`.")
+            return
+
+        # Normalize k mesh from grid sampling
+        self.points = (
+            self.resolve_points(logger) if self.points is None else self.points
+        )
+
+        # Calculate k_line_density for data quality measures
+        model_systems = self.m_xpath("m_parent.m_parent.model_system", dict=False)
+        self.k_line_density = (
+            self.resolve_k_line_density(model_systems, logger)
+            if self.k_line_density is None
+            else self.k_line_density
+        )
+
+
+class FrequencyMesh(Mesh):
+    """
+    A base section used to specify the settings of a sampling mesh in the frequency real or imaginary space.
+    """
```

Comment:
I think here `FrequencyMesh` represents any discrete mesh for [ω0, ω1, ω2, ..., ωN]. It only represents a discrete mesh whose density might be relevant to know the precision of a calculation.

Reply:
Yes, but I was more so referring to the physical interpretation. There still is one, since you are using energy units for `points`. In the context of time, frequency is Hz, not energy. There is of course a deep connection between both, as time-symmetry leads to energy conservation. QP (or at least Green's functions) implicitly uses this connection. For statistical systems (glass transitions, chemical reactions), you could recover a different time-energy relation via ergodicity, but there it's safer to just speak of s or Hz without the energy connotation.
In the case of FFT, then this mesh units could be anything really, e.g. inverted electron density in VASP.

My comment here is about whether we should reserve a generic name like `Frequency` for solely QP? I think you can retain this section, but that it should be renamed to better clarify its semantics.